src/main/java/org/assertj/jodatime/api/LocalDateAssert.java

```diff
@@ -0,0 +1,407 @@
+import static org.assertj.jodatime.error.ShouldBeBefore.shouldBeBefore;
+import static org.assertj.jodatime.error.ShouldBeBeforeOrEqualsTo.shouldBeBeforeOrEqualsTo;
+import static org.assertj.jodatime.error.ShouldHaveDayOfMonthEqualTo.shouldHaveDayOfMonthEqualTo;
+import static org.assertj.jodatime.error.ShouldHaveMonthOfYearEqualTo.shouldHaveMonthOfYearEqualTo;
+import static org.assertj.jodatime.error.ShouldHaveYearEqualTo.shouldHaveYearEqualTo;
+
+/**
+ * Assertions fot Joda {@link LocalDate} type.
+ *
+ * @author Evgenii Strepetov
+ */
+public class LocalDateAssert extends AbstractAssert<LocalDateAssert, LocalDate> {
+
+  public static final String NULL_LOCAL_DATE_PARAMETER_MESSAGE = "The LocalDate to compare actual with should not be null";
+
+  /**
+   * Creates a new <code>{@link LocalDateAssert}</code>.
+   *
+   * @param selfType the "self type"
+   * @param actual the actual value to verify
+   */
+  protected LocalDateAssert(Class<LocalDateAssert> selfType, LocalDate actual) {
+    super(actual, selfType);
+  }
+
+  // visible for test
+  protected LocalDate getActual() {
+    return actual;
+  }
+
+  /**
+   * Verifies that the year of the actual {@code LocalDate} is equal to the given year
+   * <p>
+   * Example :
+   * <pre><code class='java'> assertThat(new LocalDate(2018,3,4)).hasYear(2018);</code></pre>
+   *
+   * @param expectedYear the given year.
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code LocalDate} is {@code null}.
+   * @throws AssertionError if the year of the actual {@code LocalDate} is not equal to the given year.
+   */
+  public LocalDateAssert hasYear(int expectedYear) {
+    isNotNull();
+    if (actual.getYear() != expectedYear)
+      throw Failures.instance().failure(info, shouldHaveYearEqualTo(actual, actual.getYear(), expectedYear));
+    return this;
+  }
+
+  /**
+   * Verifies that the month of the actual {@code LocalDate} is equal to the given month
+   * <p>
+   * Example :
+   * <pre><code class='java'> assertThat(new LocalDate(2018,3,4)).hasMonthOfYear(3);</code></pre>
+   *
+   * @param expectedMonthOfYear the given month, from 1 to 12
+   * @return this assertion object.
+   * @throws AssertionError if the actual {@code LocalDate} is {@code null}.
+   * @throws AssertionError if the month of the actual {@code LocalDate} is not equal to the given month.
+   */
+  public LocalDateAssert hasMonthOfYear(int expectedMonthOfYear) {
+    isNotNull();
+    if (actual.getMonthOfYear() != expectedMonthOfYear)
+      throw Failures.instance()
+                    .failure(info, shouldHaveMonthOfYearEqualTo(actual, actual.getMonthOfYear(), expectedMonthOfYear));
+    return this;
+  }
+
+  /**
+   * Verifies that the month of the actual {@code LocalDate} is equal to the given month
+   * <p>
+   * Example :
+   * <pre><code class='java'> assertThat(new LocalDate(2018,1,1)).hasDayOfMonth(1);</code></pre>
```

Comment:
let's use a month != 1 to make it crystal clear that we check the day of month and not the month.

Reply:
Fixed